tests/unit/resource/throttle_timed.cpp

```diff
@@ -100,37 +95,52 @@ int hpx_main(int argc, char* argv[])
     return hpx::finalize();
 }
 
-int main(int argc, char* argv[])
+template <typename Scheduler>
+void test_scheduler(int argc, char* argv[])
 {
     std::vector<std::string> cfg =
     {
         "hpx.os_threads=4"
     };
 
-    using hpx::resource::scheduling_policy;
+    hpx::resource::partitioner rp(argc, argv, std::move(cfg));
 
-    std::vector<scheduling_policy> const policies =
-    {
-        scheduling_policy::local,
-        scheduling_policy::local_priority_fifo,
-        scheduling_policy::local_priority_lifo,
-        // NOTE: Static scheduling policies do not support suspending the own
-        // worker thread because they do not steal work.
-        //scheduling_policy::static_,
-        //scheduling_policy::static_priority,
-        scheduling_policy::abp_priority,
-        scheduling_policy::hierarchy,
-        //scheduling_policy::periodic_priority,
-    };
+    rp.create_thread_pool("default",
+        [](hpx::threads::policies::callback_notifier& notifier,
+            std::size_t num_threads, std::size_t thread_offset,
+            std::size_t pool_index, std::string const& pool_name)
+        -> std::unique_ptr<hpx::threads::detail::thread_pool_base>
+        {
+            typename Scheduler::init_parameter_type init(num_threads);
+            std::unique_ptr<Scheduler> scheduler(new Scheduler(init));
 
-    for (auto policy : policies)
-    {
-        // Set up the resource partitioner
-        hpx::resource::partitioner rp(argc, argv, std::move(cfg));
-        rp.create_thread_pool("default", policy);
+            auto mode = hpx::threads::policies::scheduler_mode(
+                hpx::threads::policies::do_background_work |
+                hpx::threads::policies::reduce_thread_priority |
+                hpx::threads::policies::delay_exit |
+                hpx::threads::policies::enable_elasticity);
 
-        HPX_TEST_EQ(hpx::init(argc, argv), 0);
-    }
+            std::unique_ptr<hpx::threads::detail::thread_pool_base> pool(
+                new hpx::threads::detail::scheduled_thread_pool<Scheduler>(
+                    std::move(scheduler), notifier, pool_index, pool_name, mode,
+                    thread_offset));
+
+            return pool;
+        });
+
+    HPX_TEST_EQ(hpx::init(argc, argv), 0);
+}
+
+int main(int argc, char* argv[])
+{
+    // NOTE: Static schedulers do not support suspending the own worker thread
+    // because they do not steal work. Periodic priority scheduler not tested
+    // because it does not take into account scheduler states when scheduling
+    // work.
+
```

Comment:
I know we're going to remove those schedulers. Could we add at lest an assert to those until then, making sure nobody uses them in any undue manner?